clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -259,7 +259,7 @@ public void reenqueue(RecordBatch batch, long now) {
         batch.lastAttemptMs = now;
         batch.lastAppendTime = now;
         batch.setRetry();
-        Deque<RecordBatch> deque = dequeFor(batch.topicPartition);
+        Deque<RecordBatch> deque = getOrCreateDeque(batch.topicPartition);
```

Comment:
Since `reenqueue` is public, I felt a little more uncomfortable changing the behavior here. If the class could guarantee that record batches were only created within this class (which is true at the moment), then it would seem a little safer. Maybe it makes more sense to raise an exception if there is no deque for the associated partition?